driver-core/src/main/java/com/datastax/driver/core/DirectedGraph.java

```diff
@@ -33,8 +33,10 @@
     final Map<V, Integer> vertices;
     final Multimap<V, V> adjacencyList;
     boolean wasSorted;
+    final Comparator<V> comparator;
 
-    DirectedGraph(List<V> vertices) {
+    DirectedGraph(Comparator<V> comparator, List<V> vertices) {
```

Comment:
I considered having the `Comparator` be optional, but since this is a internal class that is only used for this one purpose, figured i'd leave it as is.